### PR TITLE
Iss237: Add installation of external dependencies

### PR DESCRIPTION
**Pull request by JeremyMcCormick**
Add cmake scripts for installation of the following:

- [hps-fieldmap data](https://github.com/JeffersonLab/hps-fieldmaps)
- LCDD detector data files (from hps-java)
- [LCIO](https://github.com/jeffersonlab/hps-lcio) C++ shared library and utilities
- LCIO executable jar file
- [HPS Java](https://github.com/jeffersonlab/hps-java) executable jar file
- Local conditions file (sqlite3)

The installation of these projects is controlled by the following options, all of which are OFF by default:

- ENABLE_INSTALL_FIELDMAPS - install the magnetic field maps
- ENABLE_INSTALL_LCIO - install the LCIO bin jar and C++ tools/libraries
- ENABLE_INSTALL_HPSJAVA - install HPS Java and detector data
- ENABLE_INSTALL_CONDITIONS - install a local conditions db file for use with SQLite

The `hps-conditions-backup` project containing conditions data is private to the HPS group on github, which may cause  problems if github SSH keys are not installed locally. (The installation may hang and wait for a github username/password.)  

The external dependencies are installed under `CMAKE_INSTALL_PREFIX` in the standard way, e.g. binary executables are installed to `bin`, libraries to `lib`, etc.

The `share` directory contains the detector data, field maps, and local conditions files.

----

Additional changes included in this PR:

- Update the cmake macros for finding Python (previous way used a deprecated macro)
- Cleanup and organize the `.gitignore` files
- Find Maven when configuring the project
- Update job components in `tools.py` to use default external software locations if config for them has not been set
- Formatting/spacing in tools.py
- Remove the `enable_ptags` flag in Job (this is always on now, as ptag handling should not be optional)
- Update the environment scripts with new variables so that resources can be found at runtime (for LCIO bin jar, etc.)
- Add the `build.sh` script which executes the full CMake-based build

----

Using this branch, I have successfully tested the tritrig example scripts for a single job and local batch pool (this exercises the toolchain from generation through recon).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,7 +23,7 @@ To build the project:
 
 ```
 cd hps-mc; mkdir build; cd build
-cmake -DCMAKE_INSTALL_PREFIX=../install ..
+cmake -DCMAKE_INSTALL_PREFIX=../install -DGSL_ROOT_DIR=$GSL_ROOT_DIR ..
```
Comment (cbravo135) [inline README.md:26]: Please include a line somewhere as an example of where you set this. Is it just the directory you get from unzipping the tarball?

Comment (JeremyMcCormick) [reply README.md:26]: I updated the README just now. This is just the installation prefix of GSL (from the `--prefix` option when it was installed).

Comment (cbravo135) [reply README.md:26]: thx

**Review by cbravo135 — APPROVED** [review]
Ok, I haven't seen any major issues after using this for a day to make a few things.

**Comment by cbravo135** [comment]
I know @JeremyMcCormick wanted to do a few more tests, so I will let him click the merge button.

**Pull request merged**